src/api/student/student.service.ts

```diff
@@ -106,4 +107,112 @@ export class StudentService extends APIService<
 
     return studentPathInstance;
   }
+
+  async finishExams(user: Payload, createUserDto: FinishStudentQuiz): Promise<QuizInstance> {
+    const { fullMark, mark, answeredOptions, quizzId } = createUserDto;
+    const quizInstance = await this.db.quizInstance.findUnique({
+      where: {
+        id: quizzId,
+      },
+      include: {
+        quizStudents: true,
+      },
+    });
+    const quizStudent = quizInstance?.quizStudents.find((student) => student.studentId === user.sub);
+
+    if (!quizStudent) {
```

Comment:
this is wrong :) 
normally the student will do the exam only one time and some time will do it 2 times
so if we didn't find the student then we need to create it